Tasks/AzureVmssDeployment/operations/VirtualMachineScaleSet.ts

```diff
@@ -61,22 +42,120 @@ export default class VirtualMachineScaleSet {
                                     "commandToExecute": this.taskParameters.customScriptCommand
                                 }
                             }
-                        }
+                        };
+
+                        this._getExistingCustomScriptExtension(client, resourceGroupName, customScriptExtension, (error, matchingExtension, request, response) => {
+                            // if extension already exists, remove it
+                            if(!!matchingExtension) {
+                                this._deleteAndInstallCustomScriptExtension(client, resourceGroupName, matchingExtension, customScriptExtension, resolve, reject, (error, matchingExtension, request, response) => {
+                                    this._updateImageInternal(client, resourceGroupName, resolve, reject);
+                                });
+                            } else {
+                                this._installCustomScripExtension(client, resourceGroupName, customScriptExtension, resolve, reject, (error, result, request, response) => {
+                                    this._updateImageInternal(client, resourceGroupName, resolve, reject);
+                                });
+                            }
+                        });
+                    } else {
+                        this._updateImageInternal(client, resourceGroupName, resolve, reject);
                     }
 
-                    client.virtualMachineScaleSets.updateImage(resourceGroupName, this.taskParameters.vmssName, this.taskParameters.imageUrl, customScriptExtension, null, (error, result, request, response) => {
-                        if (error) {
-                            return reject(tl.loc("VMSSImageUpdateFailed", utils.getError(error)));
-                        }
-                        console.log(tl.loc("UpdatedVMSSImage"));
-                        return resolve();
-                    });
                     break;
                 }
             });
         });
     }
 
+    private _getResourceGroupForVmss(client, resolve, reject, callback): void {
```

Comment:
You are right; however I think that will improve the code readability. For example your main method will not have nested statements and be more readable and less error-prone like following.
var resourceGroupDetails = await this.getResourceGroupName;
var extensionStatus = await this.getExistingExtensions();